app/src/main/java/ca/llamabagel/transpo/ui/search/SearchAdapter.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2019 Derek Ellis. Subject to the MIT license.
+ */
+
+package ca.llamabagel.transpo.ui.search
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import ca.llamabagel.transpo.ui.search.viewholders.BaseAdapterItem
+
+class SearchAdapter(private val list: List<BaseAdapterItem>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = BaseAdapterItem.getViewHolder(parent, viewType)
+
+    override fun getItemCount() = list.size
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, pos: Int) = list[pos].bindViewHolder(holder)
```

Comment:
Instead of binding a `ViewHolder` to a `BaseAdapterItem`, it makes more sense ("idiomatically") to bind a `BaseAdapterItem` to a `ViewHolder`. This is especially true if using databinding (which it also should!)
```suggestion
    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, pos: Int) = holder.bind(list[pos])
```